test/tests/functional/pbs_qrun.py

```diff
@@ -115,11 +115,13 @@ def test_qrun_hangs(self):
             else:
                 self.server.submit(j)
         self.logger.info("Submitted 500 jobs with different walltime")
-        now = int(time.time())
         self.server.manager(MGR_CMD_SET, SERVER,
                             {'scheduling': 'True'})
         self.server.manager(MGR_CMD_SET, SERVER,
                             {'scheduling': 'False'}, expect=True)
+        now = int(time.time())
+        self.scheduler.log_match("Leaving Scheduling Cycle", starttime=now)
```

Comment:
I'm not sure I understand how this will work. Before setting 'now', you make sure that the scheduling cycle is finished. So, there shouldn't be any "Leaving Scheduling Cycle" message in sched logs since 'now' right?

Reply:
Thanks @agrawalravi90  for review.
self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'}, expect=True), will make sure in server the "scheduling" attribute is set false. But in some platform, in scheduler log the current cycle is still going on, so the log is coming after we set the server attribute.
But yes, it will be better if time "now" set before we set scheduling false.

Reply:
Actually, PTL does more than just check that 'scheduling' is set to False. It also checks that the server state is 'idle' before giving control back to the test. This means that by the time the test executes  "now = int(time.time())", the server should already be in the 'idle' state, meaning that the sched cycle should be finished.

Reply:
Yes @bhroam 
If we see the below log, even though in server "scheduling= false/ expect=True", still the scheduler cycle is not ended according to the sched log.

2019-01-21 02:07:53,242 INFOCLI a01: /opt/pbs/bin/qmgr -c set server scheduling=True
2019-01-21 02:07:53,280 INFO manager on a01: set server {'scheduling': 'False'}
2019-01-21 02:07:53,280 INFOCLI a01: /opt/pbs/bin/qmgr -c set server scheduling=False
2019-01-21 02:07:53,371 INFOCLI a01: /opt/pbs/bin/qstat -Bf a01.pbspro.com
**2019-01-21 02:07:53,406 INFO expect on server a01: scheduling set False && server_state set Idle server a01.pbspro.com ... OK**
**2019-01-21 02:07:53,414 INFO scheduler a01 log match: searching for "Leaving Scheduling Cycle" - No match
2019-01-21 02:07:54,015 INFO scheduler a01 log match: searching for "Leaving Scheduling Cycle"... OK**
2019-01-21 02:07:54,018 INFO run on a01: 499.a01